### ISSUE TITLE
Not compatible with Magento 2.4.4

### ISSUE THREAD

**Issue by reense** [mailcampaigns/Magento2Connector#20]
Hello,

When i try to install this module into a Magento 2.4.4 project, I get the following error:

```
Fatal error: Declaration of MailCampaigns\Magento2Connector\Model\Logger::addRecord($level, $message, array $context = Array) must be compatible with Monolog\Logger::addRecord(int $level, string $message, array $context = Array): bool in /Users/reense/Projects/xxx/vendor/mailcampaigns/magento2connector/Model/Logger.php on line 47
```

It looks like Monolog has added a return type to the addRecord() method.

**Comment by m-a-x-s-e-e-l-i-g** [comment]
Hi Reénse,

This issue is resolved in https://github.com/mailcampaigns/Magento2Connector/releases/tag/2.2.0-RC17

**Comment by reense** [comment]
@m-a-x-s-e-e-l-i-g please change the php-requirement in the composer.json file as well. We cannot use this package on a PHP 8.1 Magento instance.

**Comment by m-a-x-s-e-e-l-i-g** [comment]
@reense Done
`--ignore-platform-reqs` is also an option ;)
or
`--ignore-platform-req=php` in composer2

**Comment by reense** [comment]
@m-a-x-s-e-e-l-i-g your update still hasn't fixed the issue:
```php
Warning: Declaration of MailCampaigns\Magento2Connector\Model\Logger::addRecord(int $level, string $message, array $context = Array): bool should be compatible with Monolog\Logger::addRecord(int $level, string $message, array $context = Array, ?Monolog\DateTimeImmutable $datetime = NULL): bool in /Users/reense/Projects/xxx/vendor/mailcampaigns/magento2connector/Model/Logger.php on line 47
```

**Comment by m-a-x-s-e-e-l-i-g** [comment]
@reense 
The fatal error has been resolved. 
You probably upgraded your Monolog to another version.. What version of monolog are you currently on?
With Monolog 2.7.0 I don't experience any issues.

Also, feel free to submit a PR to fix the warning mentioned above.

**Comment by reense** [comment]
@m-a-x-s-e-e-l-i-g i know you can ignore the requirement flags! But I thought updating the constraint would be better for all of humanity 👍 

Hmm, odd. I'm also using monolog version 2.7.0 on Magento 2.4.4. I will submit a PR to fix the issue! 🙂

**Comment by reense** [comment]
@m-a-x-s-e-e-l-i-g here it is: https://github.com/mailcampaigns/Magento2Connector/pull/22

**Comment by m-a-x-s-e-e-l-i-g** [comment]
Hi @reense,
Je commit is merged en opnieuw in [2.2.0-RC17](https://github.com/mailcampaigns/Magento2Connector/releases/tag/2.2.0-RC17) released.

**Comment by twanhaverkamp** [comment]
Hi @reense,

The `$datetime` parameter of the `addRecord` method isn't nullable according to Monolog's Logger 2x file: https://github.com/Seldaek/monolog/blob/2.7.0/src/Monolog/Logger.php

2.2.0-RC18 is available for Magento 2.4.4
